components/layout/flow.rs

```diff
@@ -1582,3 +1582,67 @@ impl OpaqueFlow {
         }
     }
 }
+
+// https://drafts.csswg.org/css2/visudet.html#min-max-widths
+// https://drafts.csswg.org/css2/visudet.html#min-max-heights
+/// A min or max constraint
+///
+/// A `max` of `None` is equivalent to no limmit for the size in the given
+/// dimension. The `min` is >= 0, as negative values are illegal and by
+/// default `min` is 0.
+#[derive(Debug)]
+pub struct MinMaxConstraint {
```

Comment:
I might just stick this in `model.rs`, or else in `components/util`. It doesn't really have much to do with flows.